### ISSUE TITLE
Allow type and value with the same name

### ISSUE THREAD

**Issue by Conaclos** [facebook/flow#8992]
## Proposal and use-cases

**Current state**: Flow may bind a symbol to:
-   a type, e.g.
    ```ts
    type A = {}
    ```
-   a value, e.g. 
    ```ts
    const a = {}
    ```
-   or an entity that is both a type and a value, e.g.
    ```ts
    class C { ... }
    enum E { ... }
    ```

**Proposal**:: allow binding a same symbol to a type and a value.

This is particularly useful for opaque types:

```ts
export opaque type Percentage = number
export function Percentage(n: number): Percentage {
  assert(n > 0)
  return n
}

// in another file
import { Percentage } from ...

function process (p: Percentage) { ... }
process(Percentage(5))
```

or enum-like objects:

```ts
export type Status = 0 | 1
export const Status: { +Open: 0, +Close: 1 } = {
  Open: 0,
  Close: 1,
}

// in another file
import { Status } from ...

function process (p: Status) { ... }
process(Status.Open)
```

**This is safe because types and values are not used in the same places.**

Note that it is still disallowed to rebind a symbol for a type or a value.
The following examples are still errors:

```ts
type A = {}
class A {} //  ⚠️ A is alraedy bound
```
or 
```ts
const A = {}
class A {} // ⚠️ A is alraedy bound
```

I am using this pattern in my _TypeScript_ code base.

**Comment by SamChou19815** [comment]
Since we already have something that can both be a type or a value (classes, enums), allowing more will just make things more complicated and confusing. We will be very hesitant to accept something by Flow, only to heavily lint against it some ESLint rules to reduce confusion.

However, our current system of putting both types and values in a single namespace is problematic too. We accept some complete garbage like `const foo: foo = …` in the classic inference mode, and rejecting it under the new LTI mode for wrong reasons.

TLDR, separating the namespace for types and values is something we might want to work on. However, for this particular proposal, likely not.
